udata/core/dataset/rdf.py

```diff
@@ -164,16 +161,16 @@ def dataset_to_rdf(dataset, graph=None):
     '''
     # Use the unlocalized permalink to the dataset as URI when available
     # unless there is already an upstream URI
-    if 'uri' in dataset.extras:
-        id = URIRef(dataset.extras['uri'])
+    if dataset.harvest and dataset.harvest.uri:
+        id = URIRef(dataset.harvest.uri)
     elif dataset.id:
         id = URIRef(endpoint_for('datasets.show_redirect', 'api.dataset',
                     dataset=dataset.id, _external=True))
     else:
         id = BNode()
     # Expose upstream identifier if present
-    if 'dct:identifier' in dataset.extras:
-        identifier = dataset.extras['dct:identifier']
+    if dataset.harvest and dataset.harvest.dct_identifier:
+        identifier = dataset.harvest.dct_identifier
```

Comment:
But the spec seems quite loose on this attribute https://www.w3.org/TR/vocab-dcat-2/#Property:resource_identifier, it has to be a Literal and I think we can shove pretty much what we want in there. 

Still, we would not make sure that the identifier is indeed an identifier (which we do when we put our id). But we had the same shortfall for existing DCAT implementation, so... Maybe `remote_id` here would make sense.

Reply:
We could also consider doing something like
```
    if dataset.harvest and dataset.harvest.backend == 'DCAT':
        identifier = dataset.harvest.remote_id
```
We wouldn't need to store `dct_identifer` separately. However, it would need to know the backend harvest logic.

We can also keep with the existing solution for now.

Reply:
Existing solution seems fine.